### PR TITLE
allowDots option for stringify. Tests also updated.

### PR DESCRIPTION
**Pull request by snow01**
Added allowDots option for stringify. I have added a new test case for the same.

### CODE REVIEW

**Comment by tdzienniak** [comment]
I think that some array stringification test would be good.

**Comment by snow01** [comment]
@tdzienniak -- Following three tests are enough and expected ?

- `{ a: { b: ['c', 'd'] } }  ===> a.b[0]=c&a.b[1]=d`
- `{ a: [{ b: 'c' }] }  ===> a[0].b=c`
- `{ a: [{ b: { c: [1] } }] }  ===> a[0].b.c[0]=1`

File: lib/stringify.js [hunk 1]
```diff
@@ -65,9 +65,9 @@ internals.stringify = function (object, prefix, generateArrayPrefix, strictNullH
         }
 
         if (Array.isArray(obj)) {
-            values = values.concat(internals.stringify(obj[key], generateArrayPrefix(prefix, key), generateArrayPrefix, strictNullHandling, skipNulls, encode, filter));
+            values = values.concat(internals.stringify(obj[key], generateArrayPrefix(prefix, key), generateArrayPrefix, strictNullHandling, skipNulls, encode, filter, sort, allowDots));
```
Comment (ljharb) [inline lib/stringify.js]: it looks like you caught a minor bug/omission with the missing "sort" param here and below - i'm not sure if that was intentionally omitted or not originally. could you also add tests that cover the presence/absence of this parameter (here and below)?

Alternatively, if you could prepare a separate PR that adds the sort param and the tests for it, i'd prefer to merge that first and keep this PR focused on "allowDots" :-)

**Comment by snow01** [comment]
@ljharb - Yep, I saw 'sort' param missing as a bug due to omission. Existing tests are not capturing that bug. At depth 3 or more, without that sort does not work.

input --
`{ a: 'a', z: { zj: {zjb: 'zjb', zja: 'zja'}, zi: {zib: 'zib', zia: 'zia'} }, b: 'b' }`

Without 'sort' param fix --

`a=a&b=b&z[zi][zib]=zib&z[zi][zia]=zia&z[zj][zjb]=zjb&z[zj][zja]=zja`

With sort param fix --

`a=a&b=b&z[zi][zia]=zia&z[zi][zib]=zib&z[zj][zja]=zja&z[zj][zjb]=zjb`

Committing the test for this case in same PR, hope okay.

**Comment by ljharb** [comment]
Thanks, that's helpful. I'd still prefer a separate PR but I won't make it a sticking point :-)

**Comment by snow01** [comment]
Lint is giving error for 3 reasons -- 

- more number of parameters ===> we may need to pass parameters as opts.
- more number of statements ===> can refactor some.


/github/qs/lib/stringify.js
   23:23  error    This function has too many parameters (9). Maximum allowed is 8    max-params
   77:18  error    Function 'anonymous' has a complexity of 19                        complexity
   77:18  error    This function has too many statements (33). Maximum allowed is 32  max-statements



From PR perspective how do you suggest we do this so lint passes ?

**Comment by ljharb** [comment]
I'm fine just bumping all those limits in `.eslintrc`. I'll worry about tightening them down later.

**Comment by snow01** [comment]
Rather than bumping up values, should I change those settings to warning ?

**Comment by ljharb** [comment]
Nah, warnings are pretty useless imo since they're easy to ignore.

**Comment by snow01** [comment]
Cool. Done. 

Please pass it soon, if possible, so I can move from dependency on my branch to NPM package.

Thanks !

**Comment by ljharb** [comment]
@snow01 Released as v6.1.0

**Comment by snow01** [comment]
Thanks Jordan !

**Comment by TheSharpieOne** [comment]
Needs documentation added to the readme. Had to search the repo to find this to figure out this feature already existed and then view the changes to see how to use it.

**Comment by ljharb** [comment]
@TheSharpieOne a PR or an issue for updating the docs would be appreciated

**Pull request merged**